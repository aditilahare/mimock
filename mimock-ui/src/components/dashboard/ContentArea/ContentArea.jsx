import React from 'react';
import List from './List';
import Sidebar from './Sidebar';
import { ContentAreaContainer } from './ContentArea.style';
import PropTypes from 'prop-types';

function ContentArea({
	showSidebarSection,
	mocksListView,
	httpMethodFilter,
	handleClearFilter,
	handleOnBadgeClick,
}) {
	return (
		<ContentAreaContainer>
			<If condition={showSidebarSection}>
				<Sidebar
					onBadgeClick={handleOnBadgeClick}
					isFilterCleared={httpMethodFilter === ''}
				/>
			</If>
			<List
				mocksListView={mocksListView}
				httpMethodFilter={httpMethodFilter}
				handleClearFilter={handleClearFilter}
			/>
		</ContentAreaContainer>
	);
}

ContentArea.propTypes = {
	showSidebarSection: PropTypes.bool.isRequired,
	mocksListView: PropTypes.string.isRequired,
	httpMethodFilter: PropTypes.string.isRequired,
	handleClearFilter: PropTypes.func.isRequired,
	handleOnBadgeClick: PropTypes.func.isRequired,
};

export default ContentArea;
