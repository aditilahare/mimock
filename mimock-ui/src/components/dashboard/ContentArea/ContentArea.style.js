import tw from 'tailwind-styled-components';

export const ContentAreaContainer = tw.div`
    w-full
    flex
    xl:flex-row
    md:flex-col
    mx-auto
    mt-6
    mb-2
    justify-center
    align-middle
    px-9
    min-h-full
`;
