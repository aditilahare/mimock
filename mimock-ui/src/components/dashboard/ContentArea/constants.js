export const constants = {
	testIds: {
		contentAreaContainer: 'content-area-container',
	},
};
