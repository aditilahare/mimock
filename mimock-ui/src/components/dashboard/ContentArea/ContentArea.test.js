import React from 'react';
import { render } from '@testing-library/react';
import ContentArea from './ContentArea';

describe('ContentArea', () => {
	it('should render content area component', async () => {
		const tree = await render(<ContentArea />);

		const { container, getByTestId } = tree;

		expect(getByTestId('list-section')).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});
});
