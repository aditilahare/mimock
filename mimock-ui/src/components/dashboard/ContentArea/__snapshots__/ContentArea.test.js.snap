// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentArea should render content area component 1`] = `
<div>
  <div
    class="w-full flex flex-row mx-auto mt-6 mb-2 justify-center align-middle px-9"
  >
    <div
      class="flex flex-col justify-start align-middle w-full px-9 mt-4 mb-2"
      data-testid="list-section"
    >
      <span
        class="css-srux5c-LoaderStyle"
      >
        <span
          class="css-ilyd5x"
        />
        <span
          class="css-80tgv6"
        />
      </span>
    </div>
  </div>
</div>
`;
